app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,73 @@
+class OrderItemsController < ApplicationController
+  before_action :set_order, only: [:index, :destroy]
+
+  def index
+    @order_items = @order.order_items || []
+  end
+  # def index
+  #   @order_items = current_user.order_items # or the correct association
+  #   @OrderItem = @order_items.first # Example logic, adjust as needed
+  # end
+
+  def create
+    @order = current_user.orders.find_or_create_by(status: :cart)
```

Comment:
You are already finding the @order in set_order method. Mention the create action in `before_action :set_order, only: %i[index destroy create]`.